### PR TITLE
workload/schemachange: version gate pgvector in more places

### LINKED ISSUE #126980: roachtest: schemachange/mixed-versions failed [type "vector" does not exist when creating function]
roachtest.schemachange/mixed-versions [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16022118?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16022118?buildTab=artifacts#/schemachange/mixed-versions) on release-24.2 @ [495d20e82f09b3436f1b00d7331146fe2042dd6a](https://github.com/cockroachdb/cockroach/commits/495d20e82f09b3436f1b00d7331146fe2042dd6a):

```
(mixedversion.go:620).Run: mixed-version test failure while running step 13 (run "run schemachange workload and validation in mixed version"): full command output in run_062856.799191950_n4_COCKROACHRANDOMSEED2.log: COMMAND_PROBLEM: exit status 1
test artifacts and logs in: /artifacts/schemachange/mixed-versions/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>

See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)

See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)

See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16022118/schemachange-mixed-versions/1720679082094/1720679513799)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #126978 roachtest: schemachange/mixed-versions failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-23.1.24-rc release-blocker]
- #126883 roachtest: schemachange/mixed-versions failed [C-test-failure O-roachtest O-robot T-sql-foundations branch-release-24.1]
- #126784 roachtest: schemachange/mixed-versions failed [descriptor ID generator unavailable, migration in progress, retry later] [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-release-23.1]
- #126374 roachtest: schemachange/mixed-versions failed [existing descriptors that still need to be dropped] [C-test-failure O-roachtest O-robot P-2 T-sql-foundations branch-master]
- #125379 roachtest: schemachange/mixed-versions failed [attempting to append refresh spans after the tracked timestamp has moved forward] [C-test-failure O-roachtest O-robot T-sql-queries branch-release-23.2]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*schemachange/mixed-versions.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

Jira issue: CRDB-40217

### PR DESCRIPTION
**Pull request by rafiss**
fixes https://github.com/cockroachdb/cockroach/issues/126980
Release note: None

### CODE REVIEW

**Review by fqazi — APPROVED** [review]
:lgtm:

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/127011#-:-O1XqOej92ss15ldiyK0:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @DarrylWong and @srosenberg)

File: pkg/workload/schemachange/operation_generator.go [hunk 1]
```diff
@@ -3708,7 +3708,19 @@ func (og *operationGenerator) randType(
...
+		ctx,
+		tx,
+		clusterversion.V24_2.Version())
+	if err != nil {
+		return nil, nil, err
+	}
+
 	typ := randgen.RandSortingType(og.params.rng)
+	for pgVectorNotSupported && typ.Family() == types.PGVectorFamily {
+		typ = randgen.RandSortingType(og.params.rng)
```
Comment (renatolabs) [inline pkg/workload/schemachange/operation_generator.go:3722]: Out of curiosity: is there something stopping this code from returning pgvector again?

Comment (rafiss) [reply pkg/workload/schemachange/operation_generator.go:3722]: yes, the `for` loop will keep iterating until it generates a type that is not `pgvector`

**Pull request merged**